Source/Scene/Cesium3DTileset.js

```diff
@@ -1286,309 +1470,276 @@ define([
             return;
         }
 
-        var stats = tileset._statistics;
+        var processingQueue = selectionState.processingQueue;
+        var nextQueue = selectionState.nextQueue;
+        var finalQueue = selectionState.finalQueue;
+        var selectionQueue = selectionState.selectionQueue;
 
-        var stack = scratchStack;
-        stack.push(root);
-        while (stack.length > 0) {
-            // Depth first.  We want the high detail tiles first.
-            var t = stack.pop();
-            t.selected = false;
-            t.replaced = false;
-            ++stats.visited;
+        processingQueue.length = 0;
+        nextQueue.length = 0;
+        finalQueue.length = 0;
 
-            var visibilityPlaneMask = t.visibilityPlaneMask;
-            var fullyVisible = (visibilityPlaneMask === CullingVolume.MASK_INSIDE);
+        visitTile(tileset, root, frameState, outOfCore);
+        processingQueue.push(root);
+        selectionState.done = false;
 
-            touch(tileset, t, outOfCore);
+        var processLength = 0;
+        while (!selectionState.done) {
+            selectionState.nextQueue.length = 0;
+            processSelectionQueue(tileset, frameState, selectionState, outOfCore);
```

Comment:
> Do the functions get recreated every time selectTiles is called or is this optimized?

Based on my searching, the answer is maybe...? Would be best to avoid.

Separate functions is good approach for now.